### PR TITLE
Fix innercomments

### PR DESCRIPTION
**Pull request by shaodahong**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes https://github.com/babel/babel/issues/11469
| Patch: Bug Fix?          |Yes
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

### CODE REVIEW

File: packages/babel-parser/src/parser/comments.js [hunk 1]
```diff
@@ -264,7 +264,17 @@ export default class CommentsParser extends BaseParser {
       ) {
         node.innerComments = trailingComments;
       } else {
-        node.trailingComments = trailingComments;
+        // TrailingComments maybe contain innerComments
+        const findIndex = trailingComments.findIndex(
```
Comment (kaicataldo) [inline packages/babel-parser/src/parser/comments.js:268]: Can we name this variable `firstTrailingCommentIndex`? We could probably get ride of the comment then!

File: packages/babel-parser/src/parser/comments.js [hunk 2]
```diff
@@ -264,7 +264,17 @@ export default class CommentsParser extends BaseParser {
...
+          comment => comment.end >= node.end,
+        );
+
+        if (findIndex !== -1 && findIndex !== 0) {
```
Comment (kaicataldo) [inline packages/babel-parser/src/parser/comments.js:272]: Looks like we need tests for when:

* Trailing comments _only_ contain inner comments
* When will the second half of this (`findIndex !== 0`) evaluate truthy? Can we add a test for that?

Comment (shaodahong) [reply packages/babel-parser/src/parser/comments.js]: > Trailing comments only contain inner comments

Done.

> When will the second half of this (findIndex !== 0) evaluate truthy? Can we add a test for that?

Have it already, look like

```js
class A {
  test() {}
  // this.member = 'value';
}
```

if `findIndex === 0`mean  both **trailingComments**, so we can skip

File: packages/babel-parser/test/fixtures/comments/regression/11469/output.json [hunk 3]
```diff
@@ -0,0 +1,181 @@
+{
+  "type": "File",
+  "start":0,"end":193,"loc":{"start":{"line":1,"column":0},"end":{"line":18,"column":1}},
+  "errors": [
+    "SyntaxError: Identifier 'A' has already been declared (9:6)",
```
[the remark this answers is not in the document]
Comment (shaodahong) [reply packages/babel-parser/test/fixtures/comments/regression/11469/output.json]: Thanks, I fixed it

File: packages/babel-parser/test/fixtures/comments/regression/11469/output.json [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+      "value": " this.member = 'value';",
+      "start":109,"end":134,"loc":{"start":{"line":11,"column":4},"end":{"line":11,"column":29}}
+    },
+    {
+      "type": "CommentLine",
+      "value": " this.member = 'value';",
+      "start":166,"end":191,"loc":{"start":{"line":17,"column":2},"end":{"line":17,"column":27}}
+    }
+  ]
+}
```
Comment (nicolo-ribaudo) [inline packages/babel-parser/test/fixtures/comments/regression/11469/output.json:177]: Nit: 
```suggestion
}

```

**Comment by existentialism** [comment]
@shaodahong thanks!

**Pull request merged**